### ISSUE TITLE
Refactor `wix_arch()` implementation

### ISSUE THREAD

**Issue by volks73** [volks73/cargo-wix#126]
As mentioned by @roblabla, the [`wix_arch()`](https://github.com/volks73/cargo-wix/blob/a9cce72a8f6327439baffce9a35abbf626739f28/src/lib.rs#L527) method implementation is messy and does not support custom target specifications. This is a reminder to update the implementation if and when the `rustc --print target-spec-json` command is stabilized. It currently requires a nightly compiler. Maybe a [cargo-metadata](https://crates.io/crates/cargo_metadata)-like crate would eventually be available, too?

**Comment by roblabla** [comment]
While commenting on #127 I had the sudden realisation that I'm an utter idiot. `rustc --print target-spec-json` is unnecessary, we can instead use `rustc --print cfg` to get all the information we need in stable rust:

```
roblabla@roblabla-work:~/vm-image$ rustc --print cfg --target i686-pc-windows-msvc
debug_assertions
target_arch="x86"
target_endian="little"
target_env="msvc"
target_family="windows"
target_feature="fxsr"
target_feature="sse"
target_feature="sse2"
target_os="windows"
target_pointer_width="32"
target_vendor="pc"
windows
````

**Comment by volks73** [comment]
Ha, I am learning more and more every day. I did not know about the `--print cfg` option, too. It contains all of the information we need. We should add some of these as WiX Variables, too:

- `target_env` = `TargetEnv`
- `target_vendor` = `TargetVendor`

I am not sure the rest are needed or useful.

**Comment by volks73** [comment]
This has been resolved by adding the `WixArch` type in 0da95c9180b39f1b56123b620f4d48184210746b.

Adding WiX variables is a separate issue, so I am closing this one and future discussion should be in #118.
